components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/util/SAMLSSOUtil.java

```diff
@@ -1662,6 +1725,68 @@ public static boolean isSAMLIssuerExists(String issuerName, String tenantDomain)
         } finally {
             PrivilegedCarbonContext.endTenantFlow();
         }
+        }
+
+    /**
+     * Check whether an SP with an Issuer Entity Value similar to the Issuer of SAML request exists.
+     *
+     * @param issuerName          Issuer of SAML request.
+     * @param issuerWithQualifier Issuer value saved in the registry.
+     * @param tenantDomain
+     * @return true, if a SAML SP exists in the registry with the issuer value similar to "issuerWithQualifier"
+     * and the SAML request's issuer is equal to the issuer without qualifier.
+     */
+    public static boolean isValidSAMLIssuer(String issuerName, String issuerWithQualifier, String tenantDomain)
+            throws IdentitySAML2SSOException {
+
+        if (isSAMLIssuerExists(issuerWithQualifier, tenantDomain)) {
+            SAMLSSOServiceProviderDO sp = getSAMLSSOServiceProvider(issuerWithQualifier, tenantDomain);
```

Comment:
Why we need to load SAML service provider here? can't we use persistenceManager.isServiceProviderExists(registry, issuerName)?